apis/Google.Cloud.PubSub.V1/docs/index.md

```diff
@@ -82,6 +82,139 @@ restart listening for messages with `StartAsync(...)` again. Due to the expense
+```
+
+Here `SubscriberService` is the class that implements `BackgroundService` and uses the `SubscriberClient`
+registered with the dependency injection container to handle the messages. Once the service is registered,
+it will be automatically started when the application starts and stopped when the application exits.
+A sample implementation of `SubscriberService` is shown below:
+
+```csharp
+public class SubscriberService : BackgroundService
+{
+    private readonly SubscriberClient _subscriberClient; 
+    private readonly ILogger<SubscriberService> _logger;
+
+    public SubscriberService(SubscriberClient subscriberClient,  ILogger<SubscriberService> logger)
+    {
+        _subscriberClient = subscriberClient;
+        _logger = logger;
+    }
+
+    protected override async Task ExecuteAsync(CancellationToken stoppingToken)
+    {
+        await _subscriberClient.StartAsync(async (msg, token) =>
+        {
+            _logger.LogInformation($"Received message {msg.MessageId}: {msg.Data.ToStringUtf8()}");
+            // Handle the message.
+            return SubscriberClient.Reply.Ack;
+        });
+    }
+
+    public override async Task StopAsync(CancellationToken stoppingToken)
+    {
+        await _subscriberClient.StopAsync(stoppingToken);
+    }
+}
+```
+
+During application shutdown, the `StopAsync` method of the `SubscriberService` is invoked, which in turn calls
+the `StopAsync` method of the `SubscriberClient` to gracefully shut down the client.
+
+## IAsyncDisposable
+
+Both `PublisherClient` and `SubscriberClient` implements the `IAsyncDisposable` interface,
+allowing for asynchronous resource cleanup. When the `PublisherClient` and `SubscriberClient` are registered with the `IServiceCollection`
+using the extension methods, the container will automatically invoke the `DisposeAsync` method of the client when
+the container itself is disposed, ensuring the client will be disposed automatically.
+
+For the `PublisherClient`, when the `DisposeAsync` method is called, it asynchronously waits for the time interval
+specified in the `PublisherClient.Settings.DisposeTimeout` property for the `PublisherClient` to send any pending messages before aborting
+the clean shutdown process, that may leave some locally queued messages unsent. The time interval can be customized
+by setting the `PublisherClient.Settings.DisposeTimeout` property. If no value is specified, the default value of 5 seconds is used.
+
+For the `SubscriberClient`, when the `DisposeAsync` method is called, it asynchronously waits for the time interval
+specified in the `SubscriberClient.Settings.DisposeTimeout` property for the `SubscriberClient` to acknowledge the handled messages before aborting the
+the clean stop process, that may leave some handled messages un-acknowledged. The time interval can be customized
+by setting the `SubscriberClient.Settings.DisposeTimeout` property. If no value is specified, the default value of 5 seconds is used.
```

Comment:
This section is not a child section of the Dependency injection section so it shouldn't mention dependency injection aspects. This section only discribes the mechanis of disposing, and on the DI section, you may say something like "the DI container will dispose of all services registered as singletos, see the Dispose section for more information on what that means for Publisher and Subscriber".